pandas/core/generic.py

```diff
@@ -8835,6 +8839,10 @@ def _align_series(
             if is_datetime64tz_dtype(left.index.dtype):
                 if left.index.tz != right.index.tz:
                     if join_index is not None:
+                        # GH#33671 ensure we don't change the index on
+                        #  our original Series
+                        left = left.copy(deep=False)
```

Comment:
you don't need to deep copy, just `.copy()` as that copies the *data*; here we just need to avoid mutating the index, no?

Reply:
isnt that what `deep=False` means?

Reply:
oh woa, yes, i would remove the param then, its confusing

Reply:
if you can remove here and above the deep keyword

Reply:
i prefer to be explicit about this being non-deep

Reply:
would prefer to just use copy, this is not the pattern we use; your comment is explanatory.